### ISSUE TITLE
'apt-get update' failed after running post-pve-install on proxmox (8.0)

### ISSUE THREAD

**Issue by v1-valux** [tteck/Proxmox#1676]
After running the script `post-pve-install.sh` (and selecting to "correct the sources" probably) on my proxmox 8.0 host I encountered a TASK ERROR within proxmox which says:

`stopped: command 'apt-get update' failed: exit code 100`

When trying to `apt update` manually it fails with:

`N: Repository 'Debian bookworm' changed its 'firmware component' value from 'non-free' to 'non-free-firmware'`

I added the following to `/etc/apt/sources.list`, to fix is temporarily:

`deb https://deb.debian.org/debian bookworm main non-free-firmware`

 Is there another solution?

**Comment by tteck** [comment]
Nowhere in the `post-pve-install.sh` script does it reference 'non-free' or 'non-free-firmware' repositories. So, apparently you did _not_ select "Correct Proxmox VE sources".

```
    msg_info "Correcting Proxmox VE Sources"
    cat <<EOF >/etc/apt/sources.list
deb http://ftp.debian.org/debian ${VERSION} main contrib
deb http://ftp.debian.org/debian ${VERSION}-updates main contrib
deb http://security.debian.org/debian-security ${VERSION}-security main contrib
EOF
    msg_ok "Corrected Proxmox VE Sources"
```
If you had, you would not have gotten that error

**Comment by v1-valux** [comment]
Maybe I missed something..

as I said, I added the `non-free` line only after I encountered the issue (because it was suggested for debian 12) and after indeed confirming to correct sources within the script.

before adding it to the `sources.list` file manually, it looked exactly as in the script.

I just wanted to know what I could have done wrong, to get the error message in the first place.

**Comment by tteck** [comment]
> Maybe I missed something..

Yes you did. You missed selecting "yes" to "Correct Proxmox VE sources?"

IMO, using the `deb https://deb.debian.org/debian bookworm main non-free-firmware` repository on the Proxmox host is a terrible idea.

**Comment by tteck** [comment]
Maybe you had it referenced in `/etc/apt/sources.list.d/`?

**Comment by v1-valux** [comment]
> IMO, using the deb https://deb.debian.org/debian bookworm main non-free-firmware repository on the Proxmox host is a terrible idea.

I realized that aswell and deleted it again afterwards, but thanks for the hint.

> Maybe you had it referenced in /etc/apt/sources.list.d/?

No I dind't touch those.
Just used the script after a clean install of PVE 8.0.2

> Yes you did. You missed selecting "yes" to "Correct Proxmox VE sources?"

alright believe me or not. I got the error after **confirming**, that the `sources.list` file looked like in the script (after executing it hours ago)..

Then I added the `non-free` line
and then it worked again.

After upgrading and deleting the line afterwards, everything still seemed fine.

If I got the chance, I'll try to recreate the behaviour with a clean install of 8.0.2..
